paddle/fluid/operators/slice_op.cc

```diff
@@ -291,8 +292,80 @@ class SliceOpGradMaker : public framework::SingleGradOpMaker<T> {
   }
 };
 
+// slice op double grad
+class SliceOpDoubleGrad : public framework::OperatorWithKernel {
+ public:
+  using framework::OperatorWithKernel::OperatorWithKernel;
+
+  void InferShape(framework::InferShapeContext *ctx) const override {
+    PADDLE_ENFORCE_EQ(
+        ctx->HasInput("Input"), true,
+        platform::errors::InvalidArgument("Input of slice should not be null"));
+    auto out_dims = ctx->GetInputDim("Out");
+    auto out_grad_grad_name = "DDOut";
+    if (ctx->HasOutput(out_grad_grad_name)) {
+      ctx->SetOutputDim(out_grad_grad_name, out_dims);
+    }
+  }
+  framework::OpKernelType GetExpectedKernelType(
+      const framework::ExecutionContext &ctx) const override {
+    return framework::OpKernelType(
+        OperatorWithKernel::IndicateVarDataType(ctx, "DDInput"),
+        ctx.device_context());
+  }
+  framework::OpKernelType GetKernelTypeForVar(
+      const std::string &var_name, const Tensor &tensor,
+      const framework::OpKernelType &expected_kernel_type) const override {
+    if (var_name == "StartsTensor" || var_name == "EndsTensor") {
+      return expected_kernel_type;
+    }
+    if (var_name == "StartsTensorList" || var_name == "EndsTensorList") {
+      return expected_kernel_type;
+    }
+    return framework::OpKernelType(expected_kernel_type.data_type_,
+                                   tensor.place(), tensor.layout());
+  }
+};
+
+template <typename T>
+class SliceOpDoubleGradMaker : public framework::SingleGradOpMaker<T> {
+ public:
+  using framework::SingleGradOpMaker<T>::SingleGradOpMaker;
+
+ protected:
+  std::unique_ptr<T> Apply() const override {
+    auto *bind = new T();
+    bind->SetInput("Input", this->Input("Input"));
+    if (this->HasInput("StartsTensor")) {
+      bind->SetInput("StartsTensor", this->Input("StartsTensor"));
+    }
+    if (this->HasInput("EndsTensor")) {
+      bind->SetInput("EndsTensor", this->Input("EndsTensor"));
+    }
+    if (this->HasInput("StartsTensorList")) {
+      bind->SetInput("StartsTensorList", this->Input("StartsTensorList"));
+    }
+    if (this->HasInput("EndsTensorList")) {
+      bind->SetInput("EndsTensorList", this->Input("EndsTensorList"));
+    }
+
+    bind->SetInput("Out", this->Input("Out"));
+
+    bind->SetInput("DDInput",
+                   this->OutputGrad(framework::GradVarName("Input")));
+    bind->SetOutput("DDOut", this->InputGrad(framework::GradVarName("Out")));
+
+    bind->SetAttrMap(this->Attrs());
+    bind->SetOutput("DInput", this->InputGrad("Input"));
+    bind->SetType("slice_grad_grad");
```

Comment:
Double backward of slice is the foward of slice. Maybe can reuse slice, `bind->Setype('slice')` .